server/server.go

```diff
@@ -116,7 +120,32 @@ func handleMountEvent(ctx context.Context, client *secretmanager.Client, creds c
 				Name: secret.ResourceName,
 			}
 			smMetricRecorder := csrmetrics.OutboundRPCStartRecorder("secretmanager_access_secret_version_requests")
-			resp, err := client.AccessSecretVersion(ctx, req, callAuth)
+			loc, err := locationFromSecretResource(secret.ResourceName)
+			if err != nil {
+				errs[i] = err
+				return
+			}
+
+			if len(loc) > locationLengthLimit {
+				errs[i] = fmt.Errorf("invalid location string, please check the location")
+				return
+			}
+			var secretClient *secretmanager.Client
+			if loc == "" {
+				secretClient = client
+			} else {
+				if _, ok := regionalClients[loc]; !ok {
```

Comment:
This block is executing in a go func(). Would there be any race conditions if parallel executions are in progress?

Reply:
Thanks.
Moved all the write operation on the  cache (map) are moved outside the `go func()`. Now `go func()` only uses the values from cache